### PR TITLE
Fix `string-no-newline` end positions

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Part of #5694

> Is there anything in the PR that needs further explanation?

This pull request fixes end positions for the `string-no-new-line` rule and adds test cases for warning end positions in the following rules:

- `comment-pattern`
- `no-empty-source`

### CODE REVIEW

File: lib/rules/comment-pattern/__tests__/index.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 'use strict';
 
+const { stripIndent } = require('common-tags');
```
Comment (ybiquitous) [inline lib/rules/comment-pattern/__tests__/index.js:3]: [note] Use `stripIndent` to make multi-line code readable.

File: lib/rules/string-no-newline/__tests__/index.js [hunk 2]
```diff
@@ -43,27 +47,35 @@ testRule({
 			message: messages.rejected,
 			line: 1,
 			column: 26,
+			endLine: 2,
+			endColumn: 1,
```
[the remark this answers is not in the document]
Comment (ybiquitous) [reply lib/rules/string-no-newline/__tests__/index.js]: Fixed via 0359a21e8

**Review by jeddy3 — APPROVED** [review]
Thank you!

**Comment by ybiquitous** [comment]
Changelog entry added:

> - Fixed: `string-no-newline` end positions ([#6218](https://github.com/stylelint/stylelint/pull/6218)).

**Pull request merged**